llvm/test/CodeGen/SPIRV/hlsl-intrinsics/dot4add_i8packed.ll

```diff
@@ -1,48 +1,59 @@
-; RUN: llc -O0 -mtriple=spirv32v1.3-vulkan-unknown %s -o - | FileCheck %s
+; RUN: llc -O0 -mtriple=spirv32v1.6-vulkan-unknown %s -o - | FileCheck %s --check-prefixes=CHECK,CHECK-DOT
+; RUN: llc -O0 -mtriple=spirv32-vulkan-unknown -spirv-ext=+SPV_KHR_integer_dot_product %s -o - | FileCheck %s --check-prefixes=CHECK,CHECK-DOT
+; RUN: llc -O0 -mtriple=spirv32-vulkan-unknown %s -o - | FileCheck %s --check-prefixes=CHECK,CHECK-EXP
 ; RUN: %if spirv-tools %{ llc -O0 -mtriple=spirv32v1.3-vulkan-unknown %s -o - -filetype=obj | spirv-val %}
 
 ; CHECK-DAG: %[[#int_32:]] = OpTypeInt 32 0
-; CHECK-DAG: %[[#int_8:]] = OpTypeInt 8 0
-; CHECK-DAG: %[[#zero:]] = OpConstantNull %[[#int_8]]
-; CHECK-DAG: %[[#eight:]] = OpConstant %[[#int_8]] 8
-; CHECK-DAG: %[[#sixteen:]] = OpConstant %[[#int_8]] 16
-; CHECK-DAG: %[[#twentyfour:]] = OpConstant %[[#int_8]] 24
+; CHECK-EXP-DAG: %[[#int_8:]] = OpTypeInt 8 0
+; CHECK-EXP-DAG: %[[#zero:]] = OpConstantNull %[[#int_8]]
+; CHECK-EXP-DAG: %[[#eight:]] = OpConstant %[[#int_8]] 8
+; CHECK-EXP-DAG: %[[#sixteen:]] = OpConstant %[[#int_8]] 16
+; CHECK-EXP-DAG: %[[#twentyfour:]] = OpConstant %[[#int_8]] 24
+
 ; CHECK-LABEL: Begin function test_dot
 define noundef i32 @test_dot(i32 noundef %a, i32 noundef %b, i32 noundef %c) {
 entry:
 ; CHECK: %[[#A:]] = OpFunctionParameter %[[#int_32]]
 ; CHECK: %[[#B:]] = OpFunctionParameter %[[#int_32]]
 ; CHECK: %[[#C:]] = OpFunctionParameter %[[#int_32]]
 
+; Test that we use the dot product op when capabilities allow
+
+; CHECK-DOT: %[[#DOT:]] = OpDot %[[#int_32]] %[[#A]] %[[#B]]
```

Comment:
You should also check that the capability was added to the module. I don't think it is right now.

The first checks should be:

; CHECK-DOT: OpCapability DotProductInput4x8BitPackedKHR
; CHECK-EXT: OpExtension "SPV_KHR_integer_dot_product"

Reply:
The extension should show up only if `-spirv-ext` is used.

Reply:
Right, I believe I have addressed this now. However, I am not 100% sure about correctness. Namely: the use of `addAvailableCaps(...)` in `initAvailableCapabilitiesForOpenCL`. Do I also need to add something similar for the `initAvailableCapabilitiesForVulkan`? Even when I specify using `vulkan` as the 'operating system' in the target triple, it doesn't seem to go down that code path.

Reply:
If the test is calling `initAvailableCapabilitiesForOpenCL` and not `initAvailableCapabilitiesForVulkan`, there is something wrong with the triple. For this change you should not have to change the OpenCL code. You need to use `spirv` and not `spirv32` or `spirv64`.

Reply:
You might also need to add the capabilities that are enabled by the extensions in the Vulkan code as is done for OpenCL. It might be good to more it to `initAvailableCapabilities` since it will be the same for both.